### PR TITLE
Adding no-module-factory-load

### PR DESCRIPTION
**Pull request by arthuralee**

### CODE REVIEW

File: lib/rules/no-module-factory-load.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    return {
+        // When we visit a variable declaration, either add it as a require
+        // variable, or add it to our list of other variable declarations.
+        "VariableDeclarator": function(node) {
+            // looking for require imports of moduleFactory
+            if (node.init && node.init.type === 'CallExpression' &&
+                node.init.callee.type === 'Identifier' &&
+                node.init.callee.name === 'require') {
+
+                if (node.init.arguments[0].value.indexOf('moduleFactory.js') !== -1) {
```
Comment (tuanderful) [inline lib/rules/no-module-factory-load.js]: `node.init.arguments[0].value.includes('moduleFactory.js')`

Comment (arthuralee) [inline lib/rules/no-module-factory-load.js]: done

File: lib/rules/no-module-factory-load.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            if (node.callee.object &&
+                node.callee.object.type === 'Identifier' &&
+                node.callee.object.name === moduleFactoryVarName) {
+
+                // calling moduleFactory.load
+                if (node.callee.property &&
+                    node.callee.property.type === 'Identifier' &&
+                    node.callee.property.name === 'load') {
+
+                    context.report(node.callee, 'moduleFactory.load called. Construct the module directly instead');
```
Comment (tuanderful) [inline lib/rules/no-module-factory-load.js]: maybe add a note that its deprecated. would also be useful to have a link to a wiki page going into detail about whats changed.

**Comment by tuanderful** [comment]
ship it!

**Pull request merged**